### PR TITLE
Add health check endpoint + some cleanup

### PR DESCRIPTION
**Pull request by ashkan18**
# Problem
We need health check endpoint 

Fixes: #19 

# Solution
Added `@GET api/health` which returns 
```json
{
  "healthy": true
}
```

Possibly we can do better checks by checking database connection and sidekiq queues, but for now this should be good enough.

# Other Changes
Refactored auth related code out of `ApplicationController` to new `AuthHandler`

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,39 +1,8 @@
 class ApplicationController < ActionController::API
   include ErrorHandler
+  include AuthHandler
```
Comment (joeyAghion) [inline app/controllers/application_controller.rb]: I actually think this should go into `app/controllers/concerns`.

Comment (ashkan18) [reply app/controllers/application_controller.rb]: also the `ErrorHandler`?

Comment (joeyAghion) [reply app/controllers/application_controller.rb]: Sure!

**Comment by ashkan18** [comment]
going to merge 👍

**Pull request merged**